tests/analysis/collation_token_stream_test.cpp

```diff
@@ -58,6 +58,98 @@ TEST(collation_token_stream_test, construct) {
   }
 }
 
+TEST(collation_token_stream_test, check_variant) {
```

Comment:
Please add tests for parsing different locale objects including invalid ones. 
(E.g.   R"({"locale": {"language" : "de",
                     "encoding" : true,
                     "variant" :  123232}}) )